src/modules/User/resolvers.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  stringArg, queryField, mutationField, nonNull,
+} from 'nexus/dist';
+import { compare, hash } from 'bcrypt';
+import { sign } from 'jsonwebtoken';
+import { JWT_SECRET } from '../../env';
+
+export const me = queryField('me', {
+  type: 'User',
+  resolve: (_, __, ctx) => {
+    if (!ctx.userId) {
+      throw new Error('Not authorized');
+    }
+    return ctx.prisma.user.findUnique({
+      where: {
+        id: ctx.userId,
+      },
+    });
+  },
+});
+
+export const signIn = mutationField('signIn', {
+  type: 'AuthPayload',
+  args: {
+    username: nonNull(stringArg()),
+    password: nonNull(stringArg()),
+  },
+  resolve: async (_parent, { username, password }, ctx) => {
+    const user = await ctx.prisma.user.findUnique({
+      where: {
+        username,
+      },
+    });
+    if (!user) {
+      throw new Error('user not found');
+    }
+    const passwordValid = (await compare(password, user.password || '')) || false;
+    if (!passwordValid) {
+      throw new Error('invalid password');
+    }
+    return {
+      token: sign({ userId: user.id }, JWT_SECRET),
+      user,
+    };
+  },
+});
+
+export const signUp = mutationField('signUp', {
+  type: 'AuthPayload',
+  args: {
+    user: nonNull('UserCreateInput'),
+  },
+  resolve: async (_parent, { user }, ctx) => {
+    const {
+      username, password,
+    } = user;
+    const hashedPassword = await hash(password, 10);
+
+    const created = await ctx.prisma.user.create({
+      data: {
+        username,
+        password: hashedPassword,
+      },
+    });
+
+    return {
+      token: sign({ userId: created.id }, JWT_SECRET),
```

Comment:
- How long does this signature last?
- Does [ecosystem-api](https://github.com/makerdao-ses/ecosystem-api) jwt tokens expirable?
- How to invalidate this token, ie logout?
- How to update this token when it's expired?

Reply:
- added 7d expiration as in ecosystem api
- update: sign in again.
- logging out with tokens is not straightforward:
  - original code seems to not have this functionality
  - we would have to either decrease the expiry period from 7d to something like 10 min - 1h 
  - we could keep the list of expired tokens in the memory or db:
    - on each log out add the token to the global set
    - on each log out clean up the tokens that are past expiry
    - adjust the validation of the token with the check of it not being present in the expiry set.

any preference?

Reply:
I think the appropriate way to do logout functionality is by keeping the list of valid sessions instead of invalid ones. But I also see the advantage of storing only the logged out sessions since it's less comprisable. In any case, I would in principle not implement this logic ourselves, but instead look into authentication packages like [auth.js](https://authjs.dev/) to provide us with invalidation and renewal functionalities

To summarise,
- Let's not do it in this PR
- Before implementing it, let's investigate if there is appropriate package

Reply:
Based on the [latest mock](https://github.com/makerdao-ses/switchboard-boilerplate/issues/24), I think we would need to create `session` table, where we would store session.id, expiration date, etc and than put this session.id into jwt, to later be able to mark some of those ids as `revoked` (on logout or manual revocation)